Work with Support to refine SED documentation. (#825)
freenas/freenas-docs: userguide/system.rst

.. danger:: Remember SED passwords! If the SED password is lost, SEDs
   cannot be unlocked and their data is unavailable. While it is
   possible to specify the PSID number on the label of the device with
   :command:`sedutil-cli`, doing so **erases the contents** of the
   device rather than unlock it. Always record SED passwords whenever
   they are configured or modified and store them in a secure place!

.. _Check SED Functionality:

Check SED Functionality
^^^^^^^^^^^^^^^^^^^^^^^

When SED devices are detected during system boot, %brand% checks for
configured global and device-specific passwords.

Unlocking SEDs allows a pool to contain a mix of SED and non-SED
devices. Devices with individual passwords are unlocked with their
password. Devices without a device-specific password are unlocked using
the global password.

To verify SED locking is working correctly, go to the :ref:`Shell`.
Enter :samp:`sedutil-cli --listLockingRange 0 {password} dev/{da1}`,
where *da1* is the SED and *password* is the global or individual
password for that SED. The command returns :literal:`ReadLockEnabled: 1`,
:literal:`WriteLockEnabled: 1`, and :literal:`LockOnReset: 1` for drives

> locking enabled:

Author: ```suggestion
with locking enabled:
```

* TCG Enterprise standard for newer enterprise-grade SAS devices

> The %brand% middleware implements the security capabilities of

Reviewer: We should avoid mentioning middleware as a separate component.  As William would say, it's an implementation detail.

`camcontrol <https://www.freebsd.org/cgi/man.cgi?query=camcontrol>`__
for legacy devices and
`sedutil-cli <https://www.mankier.com/8/sedutil-cli>`__
for TCG devices. When managing a SED from the command line, it is
important to use :command:`sedutil-cli` rather than camcontrol to access
the full capabilities of the device. %brand% provides the
:command:`sedhelper` wrapper script to ease SED administration from the
command line.

> By default, SED are not locked until the administrator takes ownership

Reviewer: ```suggestion
By default, SEDs are not locked until the administrator takes ownership
```

> of them. This is done by explicitly configuring a global or per-device

Reviewer: Nonspecific "this", maybe reword:
```suggestion
of them. Ownership is taken by explicitly configuring a global or per-device
```

password in the %brand% |web-ui| and adding the password to the SEDs.

A password-protected SED protects the data stored on the device
when the device is physically removed from the %brand% system. This

> allows secure disposal of the device without having to first wipe its

Reviewer: ```suggestion
allows secure disposal of the device without having to first wipe the
```

> contents. If the device is instead removed to be repurposed on another

Reviewer: The "instead removed" makes the second sentence confusing.  Well, that, and it's hard to see how "if the password is known" relates to the first sentence.  Please clarify.  Also, the second sentence could be restructured without the if/then pause: "SEDs that will be repurposed on another system can only be unlocked if the password is known."

system, it can only be unlocked if the password is known.

.. _Deploying SEDs:

Deploying SEDs
^^^^^^^^^^^^^^

> To determine which devices support SED and their device names, go to the

Reviewer: Flip sentence around to avoid if/then (to do this, do that).  Use :ref: for Shell to give them a link.
`` Run :command:`sedutil-cli --scan` in the :ref:`Shell` to detect and list devices. The second column
of the results identifies the drive type: ``

:menuselection:`Shell`
and enter :command:`sedutil-cli --scan`. In the results:

* **no** indicates a non-SED device
* **1** indicates a legacy TCG OPAL 1 device
* **2** indicates a modern TCG OPAL 2 device
* **E** indicates a TCG Enterprise device

Example:

.. code-block:: none

   root@truenas1:~ # sedutil-cli --scan
   Scanning for Opal compliant disks
   /dev/ada0  No  32GB SATA Flash Drive SFDK003L
   /dev/ada1  No  32GB SATA Flash Drive SFDK003L
   /dev/da0   No  HGST    HUS726020AL4210  A7J0
   /dev/da1   No  HGST    HUS726020AL4210  A7J0
   /dev/da10    E WDC     WUSTR1519ASS201  B925
   /dev/da11    E WDC     WUSTR1519ASS201  B925

%brand% supports setting a global password for all detected SEDs or
setting individual passwords for each SED. Using a global password for
all SEDs is strongly recommended to simplify deployment and avoid

> physically maintaining separate passwords for each SED.

Reviewer: ```suggestion
maintaining separate passwords for each SED.
```

> **Setting a global password for SEDs**

Reviewer: This should be a section, not just bold.

Go to
:menuselection:`System --> Advanced --> SED Password`
and enter the password. **Record this password and store it in a safe
place!**

Now the SEDs must be configured with this password. Go to the

> :menuselection:`Shell`

Reviewer: Use :ref: for Shell to link them to the Shell section.

and enter :samp:`sedhelper setup {password}`, where *password* is the
global password entered in
:menuselection:`System --> Advanced --> SED Password`.

:command:`sedhelper` ensures that all detected SEDs are properly

> configured to use the provided password. Example:

Reviewer: ```suggestion
configured to use the provided password:
```

.. code-block:: none

   root@truenas1:~ # sedhelper setup abcd1234
   da9			[OK]
   da10			[OK]
   da11			[OK]

Rerun :samp:`sedhelper setup {password}` every time a new SED is placed
in the system to apply the global password to the new SED.

> **Creating separate passwords for each SED**

Reviewer: Section header.

Go to
:menuselection:`Storage --> Disks`.
Click |ui-options| for the confirmed SED, then :guilabel:`Edit`.
Enter and confirm the password in the :guilabel:`SED Password` and

> :guilabel:`Confirm SED Password` fields. Disks that have a configured

Reviewer: ```suggestion
:guilabel:`Confirm SED Password` fields. Disks with a configured
```

> SED password show bullets in their row of the :guilabel:`SED Password`

Reviewer: "show bullets" is... odd. 
The menu entry should be first, because they have to go there before seeing the column.
Then:
"...are indicated by a mark in the :guilabel:`SED Password`"

column of
:menuselection:`Storage --> Disks`.

> Conversely, the rows in that column will be empty for disks that do not

Reviewer: Please don't do this.  It's the same as "alternatively", a negative that forces the reader to stop and think oppositely.  It is not needed here:
"Disks that do not support SED are not marked in this column."

"Support SED" is still an odd way to word it ("support self-encrypting devices" is misleading).  It can probably be said differently, like "support Opal encryption" or something.  We need to research this.

support SED or are unlocked using the global password.

> Now configure the SED to use the password. Go to the

Reviewer: Rewrite to avoid Now, like "The next step is to..."

> :menuselection:`Shell`

Reviewer: :ref:

and enter :samp:`sedhelper setup --disk {da1} {password}`, where *da1*
is the SED to configure and *password* is the created password from
:menuselection:`Storage --> Disks --> Edit Disks --> SED Password`.

> **Repeat this process for each detected SED and any SED deployed to the

Reviewer: This process must be repeated for each SED and any SEDs add to the system in the future.

I don't think bold is needed here.

%brand% system in the future.**

> .. danger:: It is important to remember SED passwords! Without them,

Reviewer: Use an imperative: SED passwords *must* be remembered. If the SED password is lost,

   SEDs cannot be unlocked and their data is unavailable. While it is
   possible to specify the PSID number on the label of the device with
   :command:`sedutil-cli`, doing so **erases the contents** of the
   device rather than unlock it. Always record SED passwords whenever

>    they are configured or modified and store them in a safe place!

Reviewer: s/safe/secure/

.. _Check SED Functionality:

Check SED Functionality
^^^^^^^^^^^^^^^^^^^^^^^

When SED devices are detected during system boot, the middleware
checks for configured global and device-specific passwords. Devices with

> individual passwords are unlocked with their password and any remaining

Reviewer: Split long sentence after "password".  "Any remaining" is not needed.

> devices that have no device-specific password are unlocked using the

Reviewer: Devices without a device-specific password...

> global password. No additional manual intervention is required, but

Reviewer: The second and following sentences should be split into a separate paragraph about using a mixed pool.

:command:`sedhelper unlock` is available to manually unlock all SEDs.

> Unlocking SEDs allows a pool to contain a mix of SED and non-SED

Reviewer: Is this something we want to implicitly recommend?

devices.

To verify SED locking is working correctly, go to the

> :menuselection:`Shell`. Enter

Reviewer: :ref:

:samp:`sedutil-cli --listLockingRange 0 {password} dev/{da1}`, where
*da1* is the SED and *password* is the global or individual password for
that SED. The command returns :literal:`ReadLockEnabled: 1`,

> :literal:`WriteLockEnabled: 1`, and :literal:`LockOnReset: 1` when the

Reviewer: ```suggestion
:literal:`WriteLockEnabled: 1`, and :literal:`LockOnReset: 1` for drives with
```

> SED has locking enabled. Example:

Reviewer: ```suggestion
locking enabled:
```
